lib/utils/node-module-diagnostics.ts

```diff
@@ -0,0 +1,213 @@
+import type { PackageJson as NodePackageJson } from "./resolve-node-module"
+
+interface RootPackageJson {
+  dependencies?: Record<string, string>
+  devDependencies?: Record<string, string>
+  peerDependencies?: Record<string, string>
+  optionalDependencies?: Record<string, string>
+}
+
+interface NodeModuleInfo {
+  packageName: string
+  declaredInRoot: boolean | null
+  moduleFiles: string[]
+  modulePackageJson: NodePackageJson | null
+}
+
+type ExportValue = string | Record<string, string | Record<string, string>>
+
+type PackageJsonWithDistHints = NodePackageJson & {
+  exports?: Record<string, ExportValue> | ExportValue
+}
+
+function parseJson<T>(json?: string | null): T | null {
+  if (!json) return null
+  try {
+    return JSON.parse(json) as T
+  } catch {
+    return null
+  }
+}
+
```

Comment:
one function per file, make the export name the same as the file name. Never summarize a file name- file names must match the export